src/_common/community/channel/channel-permissions.ts

```diff
@@ -0,0 +1,29 @@
+export class ChannelPermissions {
+	public static readonly ACTION_POSTING = 'posting';
+
+	allPermissions = false;
+	noPermissions = false;
+
+	perms?: { [key: string]: boolean };
```

Comment:
I think there's a new-ish Record type in TS: `perms?: Record<string, boolean>` A bit nicer to type.